demo/.devcontainer/.zshrc

```diff
@@ -0,0 +1,89 @@
+# >>> reference: https://carlosneto.dev/blog/2024/2024-02-08-starship-zsh/
+
+# list files with details
+alias ll="ls -larht"
+
+
+# set the locale of the shell
+export LANG="en_US.UTF-8"
+
+# define VSCode as the default text editor
+export EDITOR="code -w"
+
+# specify characters considered as word boundaries for command line navigation
+export WORDCHARS=""
+
+# set the location and filename of the history file
+export HISTFILE="$HOME/.zsh_history"
+
+# set the maximum number of lines to be saved in the history file
+export HISTSIZE="100000"
+export SAVEHIST="$HISTSIZE"
+
+# disable CTRL + S and CTRL + Q
+stty -ixon
+
+# enable comments "#" expressions in the prompt shell
+setopt INTERACTIVE_COMMENTS
+
+# append new history entries to the history file
+setopt APPEND_HISTORY
+
+# save each command to the history file as soon as it is executed
+setopt INC_APPEND_HISTORY
+
+# ignore recording duplicate consecutive commands in the history
+setopt HIST_IGNORE_DUPS
+
+# ignore commands that start with a space in the history
+setopt HIST_IGNORE_SPACE
+
+# >>> bindkey tip: to discovery the code of your keys, execute "$ cat -v" and press the key, the code will be printed in your shell.
+
+# use the ZLE (zsh line editor) in emacs mode. Useful to move the cursor in large commands
+bindkey -e
+
+# navigate words using Ctrl + arrow keys
+# >>> CRTL + right arrow | CRTL + left arrow
+bindkey "^[[1;5C" forward-word
+bindkey "^[[1;5D" backward-word
+
+# macosx override
+if [[ "$OSTYPE" == "darwin"* ]]; then
+  # >>> OPT + right arrow | OPT + left arrow
+  bindkey "^[^[[C" forward-word
+  bindkey "^[^[[D" backward-word
+fi
+
+# search history using Up and Down keys
+# >>> up arrow | down arrow
+bindkey "^[[A" history-beginning-search-backward
+bindkey "^[[B" history-beginning-search-forward
+
+# jump to the start and end of the command line
+# >>> CTRL + A | CTRL + E
+bindkey "^A" beginning-of-line
+bindkey "^E" end-of-line
+# >>> Home | End
+bindkey "^[[H" beginning-of-line
+bindkey "^[[F" end-of-line
+
+# navigate menu for command output
+zstyle ':completion:*:*:*:*:*' menu select
+bindkey '^[[Z' reverse-menu-complete
+
+# delete characters using the "delete" key
+bindkey "^[[3~" delete-char
+
+
+# >>> load ZSH plugin
+
+# enable kubectl plugin autocompletion
+autoload -Uz compinit
+compinit
+
+
+# start Starship prompt
+eval "$(starship init zsh)"
+
+source /home/blinky/.motd
```

Comment:
Nah, default zsh is kinda not functioning… That's a downside of zsh.
Default bash is also a pain.

That's why – for a demo/tutorial I like having a modern dev environment instead of 90s style shell :stuck_out_tongue_winking_eye: 
The demo Dockerfile is untouched, thus without such unnecessary opinionated deps – except for the nice motd :grin:

I've added a `.bashrc`, so the users can use both, see bd517b5